cpp/unit/action.cpp

```diff
@@ -79,6 +96,130 @@ void UnitAction::move_to(Unit &target, bool use_range) {
 	this->entity->invoke(cmd);
 }
 
+TargetAction::TargetAction(Unit *u, graphic_type gt, UnitReference r, coord::phys_t rad)
+	:
+	UnitAction(u, gt),
+	target{r},
+	repath_attempts{10},
+	end_action{false},
+	radius{rad} {
+	this->update_distance();
+}
+
+TargetAction::TargetAction(Unit *u, graphic_type gt, UnitReference r)
+	:
+	TargetAction(u, gt, r, adjacent_range(u)) {
+}
+
+void TargetAction::update(unsigned int time) {
+	auto target_ptr = update_distance();
+	if (!target_ptr) {
+		return;
+	}
+
+	if (this->entity->has_attribute(attr_type::gatherer)) {
```

Comment:
is there ever a use case that doesn't fall in the scheme `if (has_attribute) { get_attribute() }`? IMO it should simply be `get_attribute()` (returning a pointer or `nullptr`).